esp-hal/src/soc/efuse_field.rs

```diff
@@ -1,118 +1,124 @@
+        let bit_off = bit_start as usize;
+        let bit_end = cmp::min(bit_count as usize, bytes.len() * 8) + bit_off;
 
         let mut last_word_off = bit_off / 32;
-        let mut last_word = unsafe { block_address.add(last_word_off).read_volatile() };
+        let mut last_word = unsafe { block.address().add(last_word_off).read_volatile() };
+
         let word_bit_off = bit_off % 32;
         let word_bit_ext = 32 - word_bit_off;
-        let mut word_off = last_word_off;
 
+        let mut word_off = last_word_off;
         for bit_off in (bit_off..bit_end).step_by(32) {
-            let word_bit_len = 32.min(bit_end - bit_off);
-
             if word_off != last_word_off {
-                // read new word
+                // Read a new word:
                 last_word_off = word_off;
-                last_word = unsafe { block_address.add(last_word_off).read_volatile() };
+                last_word = unsafe { block.address().add(last_word_off).read_volatile() };
             }
 
             let mut word = last_word >> word_bit_off;
-
             word_off += 1;
 
+            let word_bit_len = cmp::min(bit_end - bit_off, 32);
             if word_bit_len > word_bit_ext {
-                // read the next word
+                // Read the next word:
                 last_word_off = word_off;
-                last_word = unsafe { block_address.add(last_word_off).read_volatile() };
-                // append bits from a beginning of the next word
+                last_word = unsafe { block.address().add(last_word_off).read_volatile() };
+                // Append bits from a beginning of the next word:
                 word |= last_word.wrapping_shl((32 - word_bit_off) as u32);
             };
 
             if word_bit_len < 32 {
-                // mask only needed bits of a word
+                // Mask only needed bits of a word:
                 word &= u32::MAX >> (32 - word_bit_len);
             }
 
-            // get data length in bytes (ceil)
+            // Represent word as a byte slice:
             let byte_len = word_bit_len.div_ceil(8);
-            // represent word as a byte slice
             let word_bytes =
-                unsafe { core::slice::from_raw_parts(&word as *const u32 as *const u8, byte_len) };
+                unsafe { slice::from_raw_parts(&word as *const u32 as *const u8, byte_len) };
 
-            // copy word bytes to output value bytes
+            // Copy word bytes to output value bytes:
             bytes[..byte_len].copy_from_slice(word_bytes);
 
-            // move read window forward
+            // Move read window forward:
             bytes = &mut bytes[byte_len..];
         }
 
-        // fill untouched bytes with zeros
+        // Fill untouched bytes with zeros:
         bytes.fill(0);
 
         unsafe { output.assume_init() }
     }
 
-    /// Read field value in a big-endian order
-    #[inline(always)]
-    pub fn read_field_be<T: AnyBitPattern>(field: EfuseField) -> T {
```

Comment:
AFAIK the MAC address is the only thing stored in BE, and it's easy enough to just read using LE and reverse the bytes, so I didn't think this function was worth keeping around. Can revert if anybody disagrees, though.